### DISCUSSION TITLE
Welcome to Dogecoin Discussions!

### DISCUSSION THREAD

**Discussion by rnicoll** [dogecoin/dogecoin#2254] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Discuss tasks which need assistance.
  * Help each other learn.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by micaelmalta** [comment]
Hello Folks, my name is Micael, I'm a software engineering manager working at Dailymotion.com in New York. 
I have 15 Years of experience now! I m a DEVOPS working on Scalability, APIs, PHP (sic!) Python, but any good developer should be language agnostic!
My work is not a work, it's a passion: so let's have fun!

**Reply by michilumin** [reply]
Thanks Micael, agree with you on the language agnosticism :)  good to hear from you here!

**Comment by rnicoll** [comment]
Hi everyone, I'm Ross, I am one of the contributors to Dogecoin Core. I've recently been trying to get 1.21 to a point where not only does it fully sync, but also all of the automated test suite passes, but am going to be focus on leadership and community guidance more going forward, hence... this discussion space!

I've been between jobs for a month, but starting a new day job on Monday, which I'm excited for. It's not blockchain related, though.

Tell me about yourself?

**Reply by rgedwill** [reply]
Hi Ross, I too am interested in the leadership/community guidance/management side of doge. I opened #2128 - I own/run a web dev team of 8 and have a few ideas from a management/vision POV but am quickly trying to get up to speed on the codebase itself. Lmk if there's anything I can contribute with otherwise I'm sure we'll meet again in a different discussion :D nice to meet you -ryan

**Reply by AbcSxyZ** [reply]
@rgedwill Did you read The Cathedral and the bazaar by Eric Raymond (or related) ? Are you already managing/working for an open source project ? 

What ideas are you thinking about  ? I'm also interested about this kind of topics, sharing your point of view could be great.

**Comment by michilumin** [comment]
Michi here. I've been working on the fees in Core and some of the UI changes (hopefully more). Main development background is in signals/RF and crypto (as in encryption, not just the currency) - 25+ years in the field, and been glued to Dogecoin since 2014. Let's get some real implementable ideas flowing here.

**Reply by TravisCondict** [reply]
Hey, I'm new and want to learn. I know of some other projects that implement RF and blockchain. Is that something that is possible to utilize with Dogecoin?

**Comment by angelcervantesa** [comment]
Hello! Angel here, I am a student at my local university and have created the Dogecoin_Mexico project to enable students and individuals like me to build upon the existing DogeCoin infrastructure. I have years of unfinished projects, but have learned from them wether that be programming, engineering, and more.

**Comment by rgedwill** [comment]
Hi all, my name is Ryan and I own/run a web development shop (Django/react FWIW) and am very interested in contributing to doge in terms of project management/tooling/team scaling etc. I'm getting up to speed on the codebase (pretty sure the last time I wrote a line of C++ was on my compilers final exam waaaaay long ago :D) but I can't wait to see how I can help get a literal dogecoin on the literal moon

**Reply by rnicoll** [reply]
Definitely helping out with keeping discussions going, helps a lot. There's a lot of people who want to help and we're not capturing that energy well.

**Reply by rgedwill** [reply]
That makes sense. I'd argue that energy people are coming with is precisely the advantage dogecoin has over competitors - but it won't last forever

**Comment by JushBJJ** [comment]
Hello, my name is Jush, I've been programming since I was 11 years old (now turning 16 this year) and I mainly specialize in python. Many years back my first language was Java and then stuck around C and Assembly for a couple of years before I switched to a higher level language which was python. In the near-future, I plan on taking bachelors of computer science with majors on datascience and machine learning. Also, currently I am (slowly) working on creating a proof-of-concept on an algorithm alternative to PoW and PoS (<a href="https://github.com/JushBJJ/Blockchain-using-Proof-of-Interaction">here</a>) based on https://github.com/dogecoin/dogecoin/issues/2059 in the hopes of it being tested and implemented into dogecoin.

**Reply by rnicoll** [reply]
Love the enthusiasm, I'm still cautious that proof of interaction is intended where for scenarios where servers are trusted, and I'm not sure how I (as a random node operator) can verify interactions with a number of other node operators who I don't trust. Not to say there isn't an answer, just something to be aware of.

**Reply by JushBJJ** [reply]
@rnicoll 
2 weeks ago I received an email from Bramas, one of the authors of the Proof of Interaction paper.

Here's some key stuff he said:

1. "_In our protocol, the network "tour" is performed between nodes that do not trust each-other in a distributed manner_."

2. "_the protocol cannot yet replace PoW or PoS, mainly because it must run in a known network i.e., the list of participant is fixed (even if it can be very large, as I think it could run between thousands of nodes) (see the conclusion for a discussion about that)_"

3. "_There are also some things that could be problematic, such as the necessity to use a deterministic (and single output) signature scheme, which is not yet directly possible with current scheme_"

4. "_and the fact that the proof is taking more space than current PoW and PoS. So our work still opens a lots of questions._"

In regards to the second quote, Bramas sent me an updated paper on which the conclusion states:
"_However, it currently works only in networks where participants are known in advance. The design of our Blockchain protocol makes it easy to propose a possible extension to remove this assumption._"

**Comment by metatron1973** [comment]
Very cool

**Comment by fmhc** [comment]
Hello, i am 20y+ into coding and thought a lot about smart contracts on doge in the past two months. I have some basic experience with Bitcoin from 2012 (green server heat reusage project / mining setup on testservers, but did never fully sync and never mined any coins), some more blockchain knowledge from 2015-17 and digged into Hyperledger setup, ETH mining, NFTs on Tezos and recently Solana / dApps. My background is business intelligence / data engineering, did a lot of data visualization but also VR/AR bridges and LoRa WAN development and variouse IoT stuff. I have looked into the bitcoin protocol recently and thought about ways of connecting dogechain with other chains and improving the fun in doge with somehow bringing NFTs or at least memes into transactions. I found some friends via twitter who are interested in similar topics, now I'm here - at the base - reporting in for duty on the doge. I know that noone wants to change the doge too much from the way it is, so its about using what is there and improving the ecosystem and usability around. Would love to get into discussions about data analytics, smart chain options, cat images on doge and I can probably help with a bit of bugfixing, testing and technical discussions here; got some python / nodejs / c experience and currently digging into go.

**Reply by khnt2x3** [reply]
Thanks!  I see alot of potential with you and the few others in this discussion.

**Comment by AbcSxyZ** [comment]
Hi, 23y old guy who ended informatic school before the lockdown, giving me programming basics. Currently, I'm trying to learn about free & open source movement, applied in software but not only (think about Wikipedia). I discovered those fields, want to learn about open source project management.
There is other related "open stuff" I'm interested in : open innovation, open content, open access, open education, open hardware, etc.

About crypto and this community, my first time was with bitcoin couple of years ago, just to buy some smelly plants at a Wall Street e-commerce  :oncoming_police_car: 
Learning about Free and Open source software (FLOSS) and related, heard about Gamestop & co in January, having in mind crypto democratization with Coinbase IPO, April happen and Doge was an intriguing phenomenon who appear to me.

I'm seeing many doge potentials (as well as important risk, like any crypto), many things to observe, to learn, and potentially to experiment !
Cryptos are an historical moment for currencies, Dogecoin is an historical moment for cryptos.

**Comment by Stevedellaire** [comment]
Windmill/solar farms = Dogecoin

**Comment by khnt2x3** [comment]
Hi!  My name is Eric - I'm not big on programming'hard-coding!  I learned quickbasic4.5 at Contra Costa Communty College here in San Pablo, CA. - I prefer PHP.  I'm here to learn my way around DogeCoin - and am looking to build my own DogeCoin Core-QT, as I have heard that the mining ability of Core-QT has been disabled, since Pool Mining has stricken it's ability to mine Solo.  I don't know if that is true, I would like some help building a Core-QT that is able to Mine, even if the hash/difficulty is too large for it to do so, as I plan to work on getting One Million Core-QT Full Nodes going, as I believe that many would dwarf the current 1500 full nodes that exist, as stated on dogechain.so and lower the hash and difficulty down enough, so that they can one again mine.  Thus, preventing Dogecoin from becoming Centralized, as well as helping many Nodes stay online and operating.

I feel that since Bitcoin's Original Developers that I contacted back in its early days have gone silent.  And, since dogecoin was created as a Joke.  There is noone interested in eithers future.  It also seems likely that these two coins, may become centralized, as their original intent of using "Spare" Computing power to process their Blockchain is obsolete, due to Software Based Miners, who are only in it for the rewards, and not for its security or decentralizatiion.  Steps must be taken and discussed to keep the original design and implementation, working.  

I don't see any information on how these cryptocoins planned to prevent Software Miners from making the Mining Ability of the Nodes obsolete.  So - I think I have a plan, and the plan calls for atleast 1.5 million nodes to dwarf the 1500 nodes currently being the only nodes running.  So, I need to make sure the Core-QT Wallet is able to earn a Reward (even if it currently isn't capable, without the processing power) - I'd also like to work on interfacing with my Personal Node via the RPC or REST API and a Web Browser.  And, I found a feature on the Hodlcoin Core-QT Wallet, called Deposit which Hodls at an interest rate, and with it, I can double 2500 coins in 75 days to approximately 5000 coins.  I'd like to integrate this feature to the Dogecoin Core-QT Wallet.  Once, all of this is incorporated, I have an interest to use a Raspberry PI 3+ with the Web Browser API and begin selling them, to get 1.5 Million Nodes up and running.

I've also talked to a few Data Centers, and not one is interested in hosting Nodes!  Information yes!  But no Nodes!  So, working that idea out and getting some information put together to get  some Data Centers updated to understand Crypto, Nodes, and Mining, would be great.  

Thanks - Glad to meet you, I'm sure!  

Hopefully I made sense!

**Comment by Much-Mining** [comment]
Much Mining here, (Or Zach for those that wanna know). I'be a nerdy tech that used to network computers turned cryptocurrency miner &(youtuber?) (It's weird saying that idk lol). 

I believe In Dogecoin and what it stands for, so here I am.
